L1Trigger/L1CaloTrigger/plugins/L1EGammaEEProducer.cc

```diff
@@ -0,0 +1,188 @@
+        if (hw_quality == 2) {
+          // we build the EM interpreted EG object
+          l1t::EGamma eg_emint = l1t::EGamma(reco::Candidate::PolarLorentzVector(
+              cl3d->iPt(l1t::HGCalMulticluster::EnergyInterpretation::EM), cl3d->eta(), cl3d->phi(), 0.));
+          eg_emint.setHwQual(4);
+          eg_emint.setHwIso(1);
+          eg_emint.setIsoEt(-1);  // just temporarily as a dummy value
+          l1EgammaBxCollection->push_back(0, eg_emint);
+          // we also prepare for the brem recovery procedure
+          selected_multiclusters.push_back(&(*cl3d));
+          auto eta_phi_bin = get_eta_phi_bin(&(*cl3d));
+          auto bucket = etaphi_bins.find(eta_phi_bin);
+          if (bucket == etaphi_bins.end()) {
+            std::vector<const l1t::HGCalMulticluster *> vec;
+            vec.push_back(&(*cl3d));
+            etaphi_bins[eta_phi_bin] = vec;
+          } else {
+            bucket->second.push_back(&(*cl3d));
+          }
+        }
+      }
+    }
+  }
+
+  // std::cout << "# of selected HGCAL multiclusters: " << selected_multiclusters.size() << std::endl;
+
+  std::sort(selected_multiclusters.begin(), selected_multiclusters.end(), compare_cluster_pt);
+  std::set<const l1t::HGCalMulticluster *> used_clusters;
+  for (auto cl3d : selected_multiclusters) {
+    if (used_clusters.find(cl3d) == used_clusters.end()) {
+      // reco::Candidate::LorentzVector mom = cl3d->p4();
+      float pt = cl3d->pt();
+      // we drop the Had component of the energy
+      if (cl3d->hOverE() != -1)
+        pt = cl3d->pt() / (1 + cl3d->hOverE());
+      reco::Candidate::PolarLorentzVector mom(pt, cl3d->eta(), cl3d->phi(), 0.);
+      reco::Candidate::PolarLorentzVector mom_eint(
+          cl3d->iPt(l1t::HGCalMulticluster::EnergyInterpretation::EM), cl3d->eta(), cl3d->phi(), 0.);
+
+      // cl3d->pt()/(1+cl3d->hOverE())
+
+      // this is not yet used
+      used_clusters.insert(cl3d);
+      auto eta_phi_bin = get_eta_phi_bin(cl3d);
+      // std::cout << " cl pt: " << cl3d->pt()
+      //           << " eta: " << cl3d->eta()
+      //           << " phi: " << cl3d->phi()
+      //           << " eta, phi bin: " << eta_phi_bin.first << "," << eta_phi_bin.second
+      //           << std::endl;
+
+      for (int eta_bin : {eta_phi_bin.first - 1, eta_phi_bin.first, eta_phi_bin.first + 1}) {
+        for (int phi_bin : {eta_phi_bin.second - 1, eta_phi_bin.second, eta_phi_bin.second + 1}) {
+          auto bucket = etaphi_bins.find(std::make_pair(eta_bin, phi_bin));
+          if (bucket != etaphi_bins.end()) {
+            // this bucket is not empty
+            for (auto other_cl_ptr : bucket->second) {
+              if (used_clusters.find(other_cl_ptr) == used_clusters.end()) {
+                if (std::abs(other_cl_ptr->eta() - cl3d->eta()) < 0.02) {
+                  if (std::abs(reco::deltaPhi(other_cl_ptr->phi(), cl3d->phi())) < 0.1) {
+                    // std::cout << "MERGE with cl pt: " << other_cl_ptr->pt()
```

Comment:
lets leave this commented code as is, useful for checks and validation.
Will clean up in future PRs.

Reply:
done